scripts/portstat

```diff
@@ -31,289 +30,180 @@ NStats = namedtuple("NStats", "rx_ok, rx_err, rx_drop, rx_ovr, tx_ok,\
-    'RFCR': 1,
-    'RJBR': 1,
-    'RDBGC0': 2,
-    'DROP_PKT_ING': 2,
-    'ROVR': 3,
-    'TPKT': 4,
-    'TERR': 5,
-    'TFCS': 5,
-    'TJBR': 5,
-    'TRPKT': 5,
-    'TDBGC0': 6,
-    'TDBGC3': 6,
-    'TDBGC5': 6,
-    'TDBGC8': 6,
-    'UCQ_DROP_PKT(0)': 6,
-    'UCQ_DROP_PKT(1)': 6,
-    'UCQ_DROP_PKT(2)': 6,
-    'UCQ_DROP_PKT(3)': 6,
-    'UCQ_DROP_PKT(4)': 6,
-    'UCQ_DROP_PKT(5)': 6,
-    'UCQ_DROP_PKT(6)': 6,
-    'UCQ_DROP_PKT(7)': 6,
-    'UCQ_DROP_PKT(8)': 6,
-    'UCQ_DROP_PKT(9)': 6,
-    'UCQ_DROP_PKT(10)': 6,
-    'UCQ_DROP_PKT(11)': 6,
-    'TOVR': 7,
-    'RBYT': 8,
-    'TBYT': 9
-}
-
-
-def cnstat_create_bcm_element(idx, netstats, bcmstats):
-    cntr = []
-    fields = [bcmstats[i] for i in [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]]
-    cntr = NStats._make(fields)
-    return dict({netstats[0]: cntr})
-
-
-def table_as_json(table):
-    output = {}
-
-    for line in table:
-        if_name = line[0]
-
-        # Build a dictionary where the if_name is the key and the value is
-        # a dictionary that holds MTU, TX_DRP, etc
-        output[if_name] = {
-            header[1] : line[1],
-            header[2] : line[2],
-            header[3] : line[3],
-            header[4] : line[4],
-            header[5] : line[5],
-            header[6] : line[6],
-            header[7] : line[7],
-            header[8] : line[8],
-            header[9] : line[9],
-            header[10] : line[10],
-            header[11] : line[11],
-            header[12] : line[12]
-            }
-    return json.dumps(output, indent=4, sort_keys=True)
-
-
-def cnstat_print(cnstat_dict, use_json):
-    table = []
-
-    for key, data in cnstat_dict.iteritems():
-        if key == 'time':
-            continue
-        table.append((key,
-                      data.rx_ok, "N/A", "N/A", data.rx_err,
-                      data.rx_drop, data.rx_ovr,
-                      data.tx_ok, "N/A", "N/A", data.tx_err,
-                      data.tx_drop, data.tx_err))
-
-    if use_json:
-        print table_as_json(table)
-
-    else:
-        print tabulate(table, header, tablefmt='simple', stralign='right')
-
-
-def ns_diff(newstr, oldstr):
-    new, old = int(newstr), int(oldstr)
-    return '{:,}'.format(new - old)
+COUNTER_TABLE_PREFIX = "COUNTERS:"
+COUNTERS_PORT_NAME_MAP = "COUNTERS_PORT_NAME_MAP"
+
+class Portstat(object):
+    def __init__(self):
+        self.db = sswsdk.SonicV2Connector()
+        self.db.connect(self.db.COUNTERS_DB)
+
+    def get_cnstat(self):
+        """
+            Get the counters info from database.
+        """
+        def get_counters(table_id):
+            """
+                Get the counters from specific table.
+            """
+            fields = [0,0,0,0,0,0,0,0,0,0]
+            for counter_name, pos in counter_bucket_dict.iteritems():
+                full_table_id = COUNTER_TABLE_PREFIX + table_id
+                fields[pos] += int(self.db.get(self.db.COUNTERS_DB, full_table_id, counter_name))
```

Comment:
can we use hmget to get all interested fields for a key in one shot?

https://redis.io/commands/hmget